Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp

```diff
@@ -742,13 +742,14 @@ namespace AzQtComponents
 
         // check that the preview on one of the top level items was clicked
         // No need to do computations on m_itemGeometry entries since we handled the expand/collapse button with the case above
+        auto idx = indexAtPos(p);
         {
-            auto idx = indexAtPos(p);
 
             if (idx.isValid())
             {
                 selectionModel()->select(idx, QItemSelectionModel::SelectionFlag::ClearAndSelect);
                 emit clicked(idx);
+                update();
```

Comment:
We shouldn't need this `update()` here. When the selection changes, the previous selection and new selection should get repainted automatically:

```
void QAbstractItemView::selectionChanged(const QItemSelection &selected, const QItemSelection &deselected)
{
    Q_D(QAbstractItemView);
    if (isVisible() && updatesEnabled()) {
        d->viewport->update(visualRegionForSelection(deselected) \| visualRegionForSelection(selected));
    }
}
```

Might be worth doing a quick check to see why that logic isn't getting triggered. This is OK as-is, but it's a little inefficient because it's going to repaint the whole (visible) thumbnail view instead of just repainting what was changed.